service/history/shard/context_impl.go

```diff
@@ -1938,6 +2018,11 @@ func newContext(
 
 	lifecycleCtx, lifecycleCancel := context.WithCancel(context.Background())
 
+	ioConcurrency := historyConfig.ShardIOConcurrency()
+	if persistenceConfig.DataStores[persistenceConfig.DefaultStore].Cassandra != nil {
+		ioConcurrency = 1
```

Comment:
maybe add a warning log about this overwrite?

Reply:
Done.